src/useCases/drive/loadSharedItems.ts

```diff
@@ -1,21 +1,85 @@
-import { DisplayableError } from '@internxt-mobile/services/common';
 import drive from '@internxt-mobile/services/drive';
-import { ListShareLinksItem } from '@internxt/sdk/dist/drive/share/types';
+import {
+  ListAllSharedFoldersResponse,
+  ListShareLinksItem,
+  SharedFiles,
+  SharedFolders,
+} from '@internxt/sdk/dist/drive/share/types';
 import { DriveFileData, DriveFolderData } from '@internxt/sdk/dist/drive/storage/types';
-import strings from 'assets/lang/strings';
+import { UseCaseResult } from '../../types';
+import errorService from '../../services/ErrorService';
 
+const ITEMS_PER_PAGE = 50;
 export type SharedLinkResult = ListShareLinksItem & { item: DriveFileData | DriveFolderData };
-export const loadSharedLinks = async () => {
+
+export type AdvancedSharedItem = SharedFolders &
+  SharedFiles & {
+    isFolder: boolean;
+    isRootLink: boolean;
+    credentials: {
+      networkPass: string;
+      networkUser: string;
+    };
+    sharingId?: string;
+    sharingType: 'public' | 'private';
+  };
+/**
+ * Gets all shared items available
+ *
+ * @returns A list of DriveItems
+ */
+export const getSharedItems = async ({
+  page,
+  shouldGetFolders,
+  shouldGetFiles,
+}: {
+  page: number;
+  shouldGetFolders: boolean;
+  shouldGetFiles: boolean;
+}): Promise<
+  UseCaseResult<{ items: (SharedFiles & SharedFolders)[]; hasMoreFiles: boolean; hasMoreFolders: boolean }>
+> => {
   try {
-    const response = await drive.share.getShareLinks();
+    const [sharedFolders, sharedFiles] = await Promise.all([
+      shouldGetFolders
+        ? drive.share.getSharedFolders({ page, perPage: ITEMS_PER_PAGE })
+        : Promise.resolve<ListAllSharedFoldersResponse>({
```

Comment:
what's this? if this is some kind of placeholder, is better to return null, otherwise we will need to maintain the placeholder and add all the data the placeholder needs in the future